src/NLog/ConfigBuilderExtensions.cs

```diff
@@ -0,0 +1,140 @@
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+using System;
+using System.Reflection;
+using NLog.Common;
+using NLog.Config;
+using NLog.Internal;
+
+namespace NLog
+{
+    /// <summary>
+    /// Extensions for configuring NLog <see cref="LogFactory"/>.
+    /// </summary>
+    public static class ConfigBuilderExtensions
+    {
+        /// <summary>
+        /// Initializes a new configuration builder for updating <see cref="LogFactory"/> configuration
+        /// </summary>
+        public static IConfigBuilder Config(this LogFactory logFactory)
+        {
+            return new ConfigBuilder(logFactory);
+        }
+
+        /// <summary>
+        /// Registers object Type transformation from dangerous (massive) object to safe (reduced) object
+        /// </summary>
+        public static IConfigBuilder RegisterObjectTransformation<T>(this IConfigBuilder configBuilder, Func<T, object> transformer)
+        {
+            var original = ConfigurationItemFactory.Default.ObjectTypeTransform;
+            ConfigurationItemFactory.Default.ObjectTypeTransform = new ObjectTypeTransformation<T>(transformer, original);
+            return configBuilder;
+        }
+
+        /// <summary>
+        /// Registers object Type transformation from dangerous (massive) object to safe (reduced) object
+        /// </summary>
+        public static IConfigBuilder RegisterObjectTransformation(this IConfigBuilder configBuilder, string objectTypeName, Func<object,object> transformer)
```

Comment:
I'm not keen on this name, but I have to think about for a proposal. 

I don't think users would understand this name without ready docs. Maybe we should include something like destructure or serialize?

Reply:
Well the method being registered takes and object and transforms it into another object. Not sure it can be much simpler without removing important information.